tests/MediaTest.php

```diff
@@ -0,0 +1,334 @@
+    {
+        $this->mockGenerateFilename();
+
+        $response = $this->actingAs(self::$admin)->json('post', '/media', ['file' => self::$file]);
+
+        $response->assertCreated();
+
+        $this->assertDatabaseHas('media', [
+            'id' => 6,
+            'name' => 'file.png',
+            'owner_id' => self::$admin->id,
+            'is_public' => false,
+            'link' => '/storage/file.png',
+        ]);
+    }
+
+    public function testCreatePublic(): void
+    {
+        $this->mockGenerateFilename();
+
+        $response = $this->actingAs(self::$user)->json('post', '/media', [
+            'file' => self::$file,
+            'is_public' => true,
+        ]);
+
+        $response->assertCreated();
+
+        $this->assertDatabaseHas('media', [
+            'id' => 6,
+            'name' => 'file.png',
+            'owner_id' => self::$user->id,
+            'is_public' => true,
+            'link' => '/storage/file.png',
+        ]);
+    }
+
+    public function testCreateCheckUrls(): void
+    {
+        $this->mockGenerateFilename();
+
+        $this->actingAs(self::$admin)->json('post', '/media', ['file' => self::$file]);
+
+        $this->assertEquals(1, Media::where('link', 'like', '/%')->count());
+    }
+
+    public function testCreateCheckResponse(): void
+    {
+        $this->mockGenerateFilename();
+
+        $response = $this->actingAs(self::$admin)->json('post', '/media', ['file' => self::$file]);
+
+        $response->assertCreated();
+
+        $this->assertDatabaseHas('media', [
+            'id' => 6,
+            'link' => '/storage/file.png',
+        ]);
+
+        Storage::disk('local')->assertExists($this->getFilePathFromUrl('file.png'));
+
+        $this->clearUploadedFilesFolder();
+    }
+
+    public function testCreateNoAuth(): void
+    {
+        $response = $this->json('post', '/media', ['file' => self::$file]);
+
+        $response->assertUnauthorized();
+    }
+
+    public function testBulkCreate(): void
+    {
+        $this->mockGenerateFilename('file1.png', 'file2.png');
+
+        $response = $this->actingAs(self::$admin)->json('post', '/media/bulk', [
+            'media' => [
+                [
+                    'file' => self::$file,
+                    'meta' => ['test1'],
+                ],
+                [
+                    'file' => self::$file,
+                    'meta' => ['test2'],
+                ],
+            ],
+        ]);
+
+        $response->assertOk();
+
+        $this->assertDatabaseHas('media', [
+            'id' => 6,
+            'name' => 'file1.png',
+            'owner_id' => self::$admin->id,
+            'meta' => "[\"test1\"]",
+            'is_public' => false,
+        ]);
+
+        $this->assertDatabaseHas('media', [
+            'id' => 7,
+            'name' => 'file2.png',
+            'owner_id' => self::$admin->id,
+            'meta' => "[\"test2\"]",
+            'is_public' => false,
+        ]);
```

Comment:
please use ModelTestState to assert table changes

Reply:
Done. I replaced all `assertDatabaseHas` with `assertChangesEqualsFixture`.